### PR TITLE
Add rocket-plugin

### PR DESCRIPTION
**Pull request by hypery2k**
See this PR for my  [rocketChat notification plugin](https://github.com/hypery2k/rocket-chat-notifier).
Already prepared the  [wiki page](https://wiki.jenkins-ci.org/display/JENKINS/RocketChat+Plugin).

### CODE REVIEW

File: permissions/plugin-rocket.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+name: "rocket-plugin"
```
Comment (daniel-beck) [inline permissions/plugin-rocket.yml]: Name should be `rocket`.

**Comment by daniel-beck** [comment]
I can't find your plugin hosting request. Did you follow the guidelines at https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins ?

**Comment by hypery2k** [comment]
thanks for the hint. I overhead the find, see here my filled hosting request: https://issues.jenkins-ci.org/projects/HOSTING/issues/HOSTING-195

**Comment by hypery2k** [comment]
So as long as I don't have the hosting request done, we're on wait here, right?

**Comment by daniel-beck** [comment]
@hypery2k Yes. I linked to this PR from the hosting request, so this gets merged when the hosting request gets implemented.

**Comment by hypery2k** [comment]
project is created: https://github.com/jenkinsci/rocket-chat-notifier-plugin

**Comment by daniel-beck** [comment]
@hypery2k Please note that the repository and artifactId should match (except the repo has a `-plugin` suffix). So we should either rename the repo `rocketchatnotifier-plugin`, or change the `artifactId` to `rocket-chat-notifier`.

CC @batmat who mentioned a rename in HOSTING-195.

Also, the YAML file name and contents (both `name` and `path`) need to be adapted accordingly.

**Comment by daniel-beck** [comment]
@hypery2k Could you please confirm you want to keep the hyphen-less version of the artifactId? Then I'll merge this and rename your plugin repo to make it match.

**Comment by hypery2k** [comment]
yes i can confirm ;)

**Comment by daniel-beck** [comment]
@hypery2k Someone's already renamed the plugin repo, I didn't notice. Sorry about that.

And I also didn't notice that the `name` in the YAML file is still `rocket`. Please note that this must match the last part of the path or it won't work. Sorry about not catching this earlier.

**Comment by hypery2k** [comment]
sorry for the confusion. I've renamed it in the YAML file too

**Comment by daniel-beck** [comment]
@hypery2k I really need to get to the PR builder for this – I'm sorry for this mess.

Please note that the user name to specify is the Jenkins LDAP account used also for JIRA and the wiki. It looks like you have specified your GitHub user name (as the user used to create the wiki page is a different one).

Also, unless you log in once to Artifactory prior to merging this change (which will enable your permissions in ~5 minutes), it will take up to one hour after your initial login to Artifactory to enable permissions. The reason is that Artifactory user database is backed by LDAP, but it only knows about the users that have logged in at least once; and we update permissions from this repo on commit, or periodically.

**Comment by hypery2k** [comment]
no problem, for my next plugin I know how it works ;)

**Comment by daniel-beck** [comment]
@hypery2k  Please note that you need to log in to Artifactory and wait for ~30 minutes before this change will be applied. See prerequisite at https://github.com/jenkins-infra/repository-permissions-updater#requesting-permissions

**Comment by hypery2k** [comment]
I can login via Web UI, but via maven  getting following error:

```
[INFO] --- maven-deploy-plugin:2.8.2:deploy (default-deploy) @ rocketchatnotifier ---
Downloading: https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/rocketchatnotifier/0.2.2-SNAPSHOT/maven-metadata.xml
Downloaded: https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/rocketchatnotifier/0.2.2-SNAPSHOT/maven-metadata.xml (769 B at 0.6 KB/sec)
Uploading: https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/rocketchatnotifier/0.2.2-SNAPSHOT/rocketchatnotifier-0.2.2-20161112.135149-2.hpi
Uploading: https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/rocketchatnotifier/0.2.2-SNAPSHOT/rocketchatnotifier-0.2.2-20161112.135149-2.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 21.366 s
[INFO] Finished at: 2016-11-12T14:51:55+01:00
[INFO] Final Memory: 50M/874M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy (default-deploy) on project rocketchatnotifier: Failed to deploy artifacts: Could not transfer artifact org.jenkins-ci.plugins:rocketchatnotifier:hpi:0.2.2-20161112.135149-2 from/to maven.jenkins-ci.org (https://repo.jenkins-ci.org/snapshots/): Failed to transfer file: https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/rocketchatnotifier/0.2.2-SNAPSHOT/rocketchatnotifier-0.2.2-20161112.135149-2.hpi. Return code is: 401, ReasonPhrase: Unauthorized. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by daniel-beck** [comment]
@hypery2k  401 Unauthorized means you're not correctly sending your credentials to Artifactory. This is different from 403 Forbidden and unrelated to this permission system.

Check this section for suggestions: https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins#HostingPlugins-Workingaroundcommonissues

If none help, consider asking on the jenkinsci-dev list for help.

Also, are you actually trying to upload a snapshot?

**Comment by hypery2k** [comment]
Thanks for the hint. Yes, I'm trying to upload a snapshot

**Pull request merged**